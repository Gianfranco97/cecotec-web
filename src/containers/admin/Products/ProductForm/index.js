import React from "react";
import { Modal, Form, Input, InputNumber, Spin } from "antd";
import PropTypes from "prop-types";
import api from "../../../../shared/api-rest";

class ProductForm extends React.Component {
  state = {
    loading: false,
  };

  handleOk = () => {
    this.myForm.submit();
  };

  save = (values) => {
    const { closeModal, selectedProduct } = this.props;
    this.setState({ loading: true }, () => {
      setTimeout(async () => {
        try {
          if (selectedProduct)
            await api.updateProduct({ id: selectedProduct.id, ...values });
          else await api.addProduct(values);

          closeModal(true);
        } catch (error) {}
      }, 1000);
    });
  };

  render() {
    const { visible, selectedProduct, closeModal } = this.props;
    const { loading } = this.state;

    return (
      <Modal
        title={
          selectedProduct
            ? `Update '${selectedProduct.name}'`
            : `Add new product`
        }
        visible={visible}
        onOk={this.handleOk}
        onCancel={() => closeModal()}
        okText={selectedProduct ? "Update" : "Submit"}
        cancelText="Cancel"
      >
        <Spin spinning={loading} delay={500}>
          <Form
            ref={(e) => {
              this.myForm = e;
            }}
            initialValues={selectedProduct}
            onFinish={this.save}
            layout="vertical"
            name="userForm"
          >
            <Form.Item name="name" label="Name" rules={[{ required: true }]}>
              <Input />
            </Form.Item>
            <Form.Item name="price" label="Price" rules={[{ required: true }]}>
              <InputNumber min={0} />
            </Form.Item>
            <Form.Item
              name="quantity"
              label="Quantity"
              rules={[{ required: true }]}
            >
              <InputNumber min={0} />
            </Form.Item>
          </Form>
        </Spin>
      </Modal>
    );
  }
}

ProductForm.propTypes = {
  closeModal: PropTypes.func.isRequired,
  visible: PropTypes.element.isRequired,
  selectedProduct: PropTypes.object,
};

export default ProductForm;
