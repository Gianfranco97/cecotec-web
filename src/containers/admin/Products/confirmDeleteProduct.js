import React from "react";
import { Modal } from "antd";
import { ExclamationCircleOutlined } from "@ant-design/icons";

const { confirm } = Modal;

function showDeleteConfirm(name, id, onOk) {
  confirm({
    title: "Are you sure delete this product?",
    icon: <ExclamationCircleOutlined />,
    content: `All '${name}' products will be deleted`,
    okText: "Yes",
    okType: "danger",
    cancelText: "No",
    onOk() {
      onOk(id)
    },
  });
}

export default showDeleteConfirm;
