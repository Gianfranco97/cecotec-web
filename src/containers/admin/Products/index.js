import React from "react";
import { List, Button } from "antd";
import { ShoppingFilled } from "@ant-design/icons";
import ProductForm from "./ProductForm";
import confirmDeleteProduct from "./confirmDeleteProduct";
import AdminLayout from "../../../components/AdminLayout";
import api from "../../../shared/api-rest";
import "./styles.scss";

class ProductsPage extends React.Component {
  state = {
    loading: true,
    showFormModal: false,
    data: [],
    selectedProduct: null,
  };

  componentDidMount() {
    this.getData();
  }

  deleteProduct = (id) => {
    this.setState({ loading: true }, async () => {
      try {
        await api.deleteProduct(id);

        this.getData();
      } catch (error) {}
    });
  };

  getData = () => {
    this.setState({ loading: true }, () => {
      setTimeout(async () => {
        try {
          const res = await api.getProducts();

          this.setState({
            loading: false,
            data: res,
          });
        } catch (error) {}
      }, 1000);
    });
  };

  closeModal = (update) => {
    if (update) this.getData();
    this.setState({ showFormModal: false, selectedProduct: null });
  };

  render() {
    const { loading, data, showFormModal, selectedProduct } = this.state;

    return (
      <AdminLayout title="Products">
        <List
          className="my-list"
          loading={loading}
          itemLayout="horizontal"
          dataSource={data}
          renderItem={(item) => (
            <List.Item
              key={`product-${item.id}`}
              actions={[
                <Button
                  type="link"
                  onClick={() => {
                    this.setState({
                      selectedProduct: item,
                      showFormModal: true,
                    });
                  }}
                >
                  edit
                </Button>,
                <Button
                  type="link"
                  onClick={() =>
                    confirmDeleteProduct(item.name, item.id, this.deleteProduct)
                  }
                >
                  delete
                </Button>,
              ]}
            >
              <List.Item.Meta
                avatar={<ShoppingFilled style={{ fontSize: "22px" }} />}
                title={item.name}
                description={
                  <>
                    <b>Price: </b> {item.price} <br />
                    <b>Quantity: </b> {item.quantity}
                  </>
                }
              />
            </List.Item>
          )}
        />

        <Button onClick={() => this.setState({ showFormModal: true })}>
          + ADD PRODUCT
        </Button>

        {showFormModal && (
          <ProductForm
            visible
            closeModal={this.closeModal}
            selectedProduct={selectedProduct}
          />
        )}
      </AdminLayout>
    );
  }
}

export default ProductsPage;
