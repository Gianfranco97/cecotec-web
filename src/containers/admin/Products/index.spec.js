import React from "react";
import { shallow } from "enzyme";
import Products from "./index.js";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

it("The snapshot matches", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Products />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(<Products />);
});
