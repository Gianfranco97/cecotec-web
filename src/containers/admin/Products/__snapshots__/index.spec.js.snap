// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The snapshot matches 1`] = `
<section
  className="ant-layout ant-layout-has-sider admin-layout"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="logo"
      >
        <img
          alt="cecotec"
          src="cecotec-logo.jpg"
        />
      </div>
      <ul
        className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
        onMouseEnter={[Function]}
        onTransitionEnd={[Function]}
        role="menu"
        style={Object {}}
      >
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <span
            aria-label="home"
            className="anticon anticon-home"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
          <span>
            Home
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <span
            aria-label="user"
            className="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
          <span>
            Clients
          </span>
        </li>
        <li
          className="ant-menu-item ant-menu-item-selected"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <span
            aria-label="unordered-list"
            className="anticon anticon-unordered-list"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="unordered-list"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M912 192H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM104 228a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <span>
            Products
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <span
            aria-label="logout"
            className="anticon anticon-logout"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="logout"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
              />
            </svg>
          </span>
          <span>
            Logout
          </span>
        </li>
      </ul>
    </div>
  </aside>
  <section
    className="ant-layout"
  >
    <header
      className="ant-layout-header site-layout-sub-header-background"
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <h1
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        Products
      </h1>
    </header>
    <main
      className="ant-layout-content admin-content"
      style={
        Object {
          "margin": "24px 16px 0",
        }
      }
    >
      <div
        className="site-layout-background"
        style={
          Object {
            "padding": 24,
          }
        }
      >
        <div
          className="ant-list my-list ant-list-split ant-list-loading"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div>
              <div
                className="ant-spin ant-spin-spinning"
              >
                <span
                  className="ant-spin-dot ant-spin-dot-spin"
                >
                  <i
                    className="ant-spin-dot-item"
                  />
                  <i
                    className="ant-spin-dot-item"
                  />
                  <i
                    className="ant-spin-dot-item"
                  />
                  <i
                    className="ant-spin-dot-item"
                  />
                </span>
              </div>
            </div>
            <div
              className="ant-spin-container ant-spin-blur"
            >
              <div
                style={
                  Object {
                    "minHeight": 53,
                  }
                }
              />
            </div>
          </div>
        </div>
        <button
          className="ant-btn"
          onClick={[Function]}
          type="button"
        >
          <span>
            + ADD PRODUCT
          </span>
        </button>
      </div>
    </main>
    <footer
      className="ant-layout-footer"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Gianfranco Manganiello Â©2020
    </footer>
  </section>
</section>
`;
